daemon/containerd/image.go

```diff
@@ -25,17 +25,46 @@ import (
 
 var errInconsistentData error = errors.New("consistency error: data changed during operation, retry")
 
+type errPlatformNotFound struct {
+	wanted   ocispec.Platform
+	imageRef string
+}
+
+func (e *errPlatformNotFound) NotFound() {}
+func (e *errPlatformNotFound) Error() string {
+	msg := "image with reference " + e.imageRef + " was found but does not match the specified platform"
```

Comment:
Perhaps `requested` instead of  `specified` would work for this? Thinking about situations where no explicit platform was requested, so the default was used (or don't we produce the error in any of those cases)?


Also considering if `match` -> `provide` (or `support`); which may be more accurate for multi-arch 🤔

Reply:
It would! But that would be different from the graphdrivers error returned in the same case. So that could break potential error string matching.

IMO, until we provide a better way for consumers to check for this error, we should stick to the existing wording.

Reply:
> IMO, until we provide a better way for consumers to check for this error, we should stick to the existing wording.

Hmm.. right; you mean distinction between "image present, but not platform" vs "image not present"? FWIW; I was more concerned about the `No such` errors, because those prefixes were used in docker itself to match error-types; for the errors we're modifying here, we never had that as a contract, so I'm much more OK with changing the error format.

Reply:
> but unfortunately Go stringly-typed errors forced that upon us implicitly I think 😅

For exported, sentinel errors, for sure (but also the reason we chose to use the "interface-matching" approach in errdefs). I think in this case, the error itself is never returned to the client as a go-error that can be matched, as the API is in between, so at most the client receives a typed error (errdefs type that is) with an error message that should be considered informational, but in itself not something to control flow.

> but I found at least one instance:

`:sad_panda:` yeah, that's not great. At least it looks like that code already calls it out for being brittle and temporary, and it looks like the issue linked to that comment was closed / addressed https://github.com/buildpacks/pack/issues/2079. That said, I agree we may have to look if the current logic of an image either being "there" or "not there" (`errdefs.NotFound`) is still sufficient to handle logic on the client side, or if we need more specific errors (that also make it accross the API).

I'm ok with keeping the error-message as-is for now, but we should indeed make sure that we don't paint ourself (again) into a corner where error-strings become an undocumented API, and we _must_ preserve error-messages (like we had with the `No such` prefix).

Reply:
Perhaps we should add a section to the API docs describing that error-message MUST be considered informational, and can change at any time, so string-matching being highly discouraged.